eyetrackprep/src/pupil2bids.py

```diff
@@ -305,6 +303,333 @@ def compile_rawfile_list(
+        with open(ip_path, 'r') as f:
+            iplayer = json.load(f)
+        sync_ts = iplayer['start_time_synced_s']
+        syst_ts = iplayer['start_time_system_s']
+        rsv = f"Pupil Capture version {iplayer['recording_software_version']}"
+
+        # check if gaze timestamp is on synced time
+        is_sync_gz = (gz0_ts-sync_ts)**2 < (gz0_ts-syst_ts)**2
+        # check if TTL 0 is on synced time
+        is_sync_ot = (on_time-sync_ts)**2 < (on_time-syst_ts)**2
+        # If gaze and TTL 0 are on different clocks, 
+        # convert TTL 0 to the same clock as the gaze timestamps
+        if is_sync_gz != is_sync_ot:
+            if is_sync_ot:
+                # convert TTL 0 from synced to system clock
+                on_time += (syst_ts - sync_ts)
+                off_time += (syst_ts - sync_ts)
+            else:
+                # convert TTL 0 from system to synced clock                
+                on_time += (sync_ts - syst_ts)
+                off_time += (sync_ts - syst_ts)
+    
+        return on_time, off_time, rsv
+    
+    else:
+        """
+        If the log file cannot be parsed, estimate the scanner onset time 
+        from the run's 12th gaze timestamp (estimated from differences between 
+        gaze timestamps and logged TTL 0 for 14 CNeuroMod tasks with eye-tracking data)
+        """        
+        log_qc(f"Warning: failed log parsing, run onset and offset time estimated from gaze timestamps for {fnum}", qc_path)
+        return gz0_ts, gzn_ts, "n/a"
+
+
+def extract_gaze(
+    seri_gaze,
+    onset_time,
+) -> list:
+    """
+    Converts nested gaze dictionary to two lists of arrays (one for BIDS, one for plotting)
+    """
+    bids_gaze_list = []
+
+    for gaze in seri_gaze:
+        gaze_timestamp = gaze['timestamp'] - onset_time
+        if gaze_timestamp > 0.0:
+            gaze_x, gaze_y = gaze['norm_pos']
+            gaze_conf = gaze['confidence']
+            pupil_data = gaze['base_data'][0]
+            pupil_x, pupil_y = pupil_data['norm_pos']
+            pupil_diameter = pupil_data['diameter']
+            ellipse_data = pupil_data['ellipse']
+            ellipse_axe_a, ellipse_axe_b = ellipse_data['axes']
+            ellipse_angle = ellipse_data['angle']
+            ellipse_center_x, ellipse_center_y = ellipse_data['center']
+
+            bids_gaze_list.append([
+                gaze_timestamp, gaze_x, gaze_y, gaze_conf, 
+                pupil_x, pupil_y, pupil_diameter, 
+                ellipse_axe_a, ellipse_axe_b, ellipse_angle, 
+                ellipse_center_x, ellipse_center_y,
+            ]) 
+                
+    return bids_gaze_list
+
+
+def detect_freezes(
+    gaze: np.array,
+    out_file: str,
+    run_duration: float,
+) -> float:
+    """
+    Identifies temporal gaps in eye-tracking data due to camera freezes, 
+    and exports them as physioevents.tsv file.
+
+    Detects instances of inter-sample interval > 0.005s (indicating a dropped frame 
```

Comment:
Based on the code, I'm assuming that this is now: `Detects instances of inter-sample interval > 0.5s`